ark/jsonschema/string.ts

```diff
@@ -0,0 +1,43 @@
+import { rootSchema, type Intersection } from "@ark/schema"
+import type { Type, string } from "arktype"
+import { JsonSchema } from "./scope.ts"
+
+export const validateJsonSchemaString = JsonSchema.StringSchema.pipe(
+	jsonSchema => {
+		const arktypeStringSchema: Intersection.Schema<string> = {
+			domain: "string"
+		}
+
+		if ("maxLength" in jsonSchema)
+			arktypeStringSchema.maxLength = jsonSchema.maxLength
+		if ("minLength" in jsonSchema)
+			arktypeStringSchema.minLength = jsonSchema.minLength
+		if ("pattern" in jsonSchema) {
+			if (jsonSchema.pattern instanceof RegExp) {
+				arktypeStringSchema.pattern = [
+					// Strip leading and trailing slashes from RegExp
+					jsonSchema.pattern.toString().slice(1, -1)
+				]
+			} else arktypeStringSchema.pattern = [jsonSchema.pattern]
+		}
+		return rootSchema(arktypeStringSchema) as unknown as Type<string>
+	}
+)
+
+export type inferJsonSchemaString<stringSchema, T = string> =
+	"maxLength" extends keyof stringSchema ?
+		inferJsonSchemaString<
+			Omit<stringSchema, "maxLength">,
+			T & string.atMostLength<stringSchema["maxLength"]>
+		>
+	: "minLength" extends keyof stringSchema ?
+		inferJsonSchemaString<
+			Omit<stringSchema, "minLength">,
+			T & string.atLeastLength<stringSchema["minLength"]>
```

Comment:
This approach of intersecting `string.atLeastLength` directly is likely what is breaking the name preservation.

As mentioned before, I'm honestly not sure if all of this inference is worth it as in the vast majority of situations where you want to create ArkType from JSON schema, it's as part of some dynamic transformation process. You'd never want to actually maintain your types in JSON schema just to translate them to arktype/`@ark/schema`, which has a richer internal structure as well as an easier definition syntax.

I know it would be cool if this worked perfectly, but if the goal is to get the most useful parts of the feature merged ASAP (which are obviously written very diligently and would provide a lot of value), the best path is probably to forego adding inference for now as I've done for `@ark/schema`. We can always go back and add it in later, but once we commit to it, it's a lot to support and at this point, it's not necessarily clear the upside is sufficient to justify it.

Reply:
Marking as resolved since this is no longer relevant now that we've removed type inference (07a5215)